doc/source/whatsnew/v0.18.1.txt

```diff
@@ -29,12 +29,16 @@ New features
 Enhancements
 ~~~~~~~~~~~~
 
+- ``Index.take`` now handles ``allow_fill`` and ``fill_value`` consistently (:issue:`12631`)
 
+  * If ``allow_fill`` is ``True`` and ``fill_value`` is not ``None``, indices specified with -1 is regarded as NA. If ``Index`` cannot hold NA, it raises ``ValueError``.
+  * Otherwise, indeces specified with -1 is regarded as the last element.
 
```

Comment:
I would remove these last 2 lines. Not necessary from a user perspective to explain more.